modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java

```diff
@@ -90,6 +94,7 @@ class S3BlobStore implements BlobStore {
     private final StatsCollectors statsCollectors = new StatsCollectors();
 
     private final int bulkDeletionBatchSize;
+    private final BackoffPolicy retryThrottledDeleteBackoffPolicy;
```

Comment:
It seems like a backoff policy is what we want here, this one comes from bulk package, but it does seem to be re-used in other non-bulk places.

Reply:
Yeah makes sense but I would rather not add more tech debt by depending on `org.elasticsearch.action.bulk`. Could you move this functionality somewhere more appropriate first, and add at least a basic test suite? Not sure exactly where would be good, maybe `o.e.common` or `o.e.common.time`?

Reply:
Done in https://github.com/elastic/elasticsearch/pull/113500